### PR TITLE
Updated Pin Field component to guard against setting non numeric values

### LINKED ISSUE #125: [Bug]: Update Pin Field to guard against setting non numeric values
### Components Version

0.5.0

### Which package type are you using?

Unbundled

### What type of bug is this?

Behaviour Error

### Test Case/Reproduction

_No response_

### Description

Currently the Pin Field only has a guard on the keydown event which will limit a user from updating the value of the Pin field via physical or virtual keyboard.

The requirement is to add the guard to the following lit functions

firstUpdated - Invoked when the element is first updated. Implement to perform one time work on the element after update.
attributeChangedCallback - Synchronizes property values when attributes change.

Suggestion is to add a event listener for paste events to ensure that only numeric values can be pasted in the Pin field.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have [searched for existing issues](https://github.com/capitec/omni-components/issues) and to the best of my knowledge this is not a duplicate

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

closes #125 

The Pin Field component required extra hardening to stop users from setting non-numeric values. 

This change was tested on multiple browsers and mobile devices

Test results 

image

(*Link to bug/feature from [Issues](../issues) and how you resolve it*)

### All Submissions:

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/pin-field/PinField.ts [hunk 1]
```diff
@@ -117,22 +117,45 @@ export class PinField extends OmniFormElement {
         }
     }
 
+    _sanitiseValue(value: string) {
+        if (value !== null && value !== undefined && value !== '') {
```
Comment (stefan505) [inline src/pin-field/PinField.ts:121]: Is this not equivalent to ` if(value) ...` ?

File: src/pin-field/PinField.ts [hunk 2]
```diff
@@ -117,22 +117,45 @@ export class PinField extends OmniFormElement {
         }
     }
 
+    _sanitiseValue(value: string) {
```
Comment (stefan505) [inline src/pin-field/PinField.ts:120]: Add some comments to the function, its at the "business" end of the component.

File: src/pin-field/PinField.ts [hunk 3]
```diff
@@ -117,22 +117,45 @@ export class PinField extends OmniFormElement {
...
-        // Stop alpha keys
-        if (e.key >= 'a' && e.key <= 'z') {
-            e.preventDefault();
-            return;
+    // Used to check if the value provided in a valid numeric value.
+    _isNumber(number: string) {
+        return /\d/.test(number);
+    }
+
+    //The type of the input is set to text which will allow alpha-numeric characters this function is to block all non numeric input values.
```
Comment (stefan505) [inline src/pin-field/PinField.ts:145]: Check grammar.

**Review by BOTLANNER — APPROVED** [review]

File: src/pin-field/PinField.ts [hunk 4]
```diff
@@ -117,22 +117,45 @@ export class PinField extends OmniFormElement {
...
+    _beforeInput(e: InputEvent) {
+        const input = this._inputElement as HTMLInputElement;
+        if (input && e.data !== null) {
```
Comment (stefan505) [inline src/pin-field/PinField.ts:148]: Can the inner `if` can be combined into a single statement?

File: src/pin-field/PinField.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/pin-field/PinField.ts]: Updated the if statement accordingly

File: src/pin-field/PinField.ts (hunk 3 above)
Comment (chromaticWaster) [reply src/pin-field/PinField.ts]: Updated the comment

**Review by stefan505 — APPROVED** [review]

**Pull request merged**